File: ProyectoCivitasJava/src/civitas/SorpresaEspeculador.java
/**
 * Proyecto Civitas
 * Sergio Hervás Cobo
 * Javier Suárez Lloréns
 * Programación y Diseño Orientado a Objetos
 * 2º B3 - Grado en Ingeniería Informática
 * E.T.S.I. Informática y de Telecomunicación
 * Universidad de Granada
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author sergio
 */
public class SorpresaEspeculador extends Sorpresa {

    private int fianza;

    SorpresaEspeculador(int cantidad) {
        super("Te conviertes en jugador especulador.");
        fianza = cantidad;
    }

    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos) {
        if (jugadorCorrecto(actual, todos)) {
            informe(actual, todos);
            todos.set(actual,new JugadorEspeculador(todos.get(actual), fianza));
        }
    }
    
    @Override
    public String toString(){
        String salida = "Jugador -> Especulador";
        return salida;
    }
}

File: ProyectoCivitasRuby/lib/modelo/sorpresa_especulador.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative 'jugador_especulador'
require_relative 'sorpresa'
module Civitas
  class SorpresaEspeculador < Sorpresa
    
    public_class_method :new 
    
    def initialize(fianza)
      super("Convertir jugador a especulador")
      @fianza = fianza
    end
    
    #@Override
    def aplicarajugador(actual, todos)
      if(jugador_correcto(actual, todos))
        informe(actual, todos)
        jugador = JugadorEspeculador.nuevo_especulador(todos[actual],@fianza)
        todos[actual] = jugador;
      end
    end
    
    #@Override
    def to_s
            salida = "{@texto}\n"
    end
  end
end